### DISCUSSION TITLE
Fill template placeholder from the API

### DISCUSSION THREAD

**Discussion by pstuifzand** [outline/outline#1716] [Ideas]
It could be possible to fill the template placeholders from the API when creating a new document from a template.

A use case would be to automatically create documents from meetings or sprints. It could be filled with information from other places and collected in one document every sprints of meeting.

**Comment by tommoor** [comment]
Hi @pstuifzand – seems totally doable, the real question is how you'd set up such a thing – how would the UI look that would enable something like this?

**Comment by pstuifzand** [comment]
I see your point about that. The placeholders need names, so they can be referenced in the API. I'm not sure how to do that.

Perhaps we can fill templates externally and create them in the normal way for now.

**Comment by RickCogley** [comment]
When you export a doc with template placeholders as markdown, they are sitting there like `!!Placeholder!!`, so, presumably you can name the placeholders something searchable and unique, pull the whole template via the API, replace the placeholders with a regex, then push it back as a new document. That way, you can do the template editing inside Outline. 

Would this work?

**Reply by tommoor** [reply]
Definitely sounds like it would work to me

**Reply by RickCogley** [reply]
Thanks @tommoor. Yeah, a quick POST to the `documents.list` endpoint  with paw shows the text string for the template has the placeholder strings I used. 

<img width="1721" alt="JRC Screenshot 2021-01-09 at 17 50 39" src="https://user-images.githubusercontent.com/512328/104087472-4f638080-52a3-11eb-9fea-9a553d01d765.png">
